### ISSUE TITLE
0.6.2 is misbehaving with SLURM

### ISSUE THREAD

**Issue by JohnUrban** [PacificBiosciences/FALCON-integrate#53]
When running 0.6.2 in SLURM mode, no error is thrown, but nothing is ever done either. It only keeps spitting out this type of message:

```
[INFO]tick: 4, #updatedTasks: 1, sleep_time=0.100000
[INFO]tick: 8, #updatedTasks: 1, sleep_time=0.500000
[INFO]tick: 16, #updatedTasks: 1, sleep_time=1.000000
[INFO]tick: 32, #updatedTasks: 1, sleep_time=1.000000
[INFO]tick: 64, #updatedTasks: 1, sleep_time=1.000000
[INFO]tick: 128, #updatedTasks: 1, sleep_time=1.000000
[INFO]tick: 256, #updatedTasks: 1, sleep_time=1.000000
```

I have been running as a batch script that looks like:

```
#!/bin/bash
#SBATCH -J falcon-synth0-1
#SBATCH -c 4
#SBATCH --qos=ccmb-condo
#SBATCH --mem=4g
#SBATCH --time=00:30:00
###########################
unset PYTHONPATH
. ~/software/falcon/FALCON-integrate/fc_env/bin/activate
fc_run.py fc_run.cfg
```

In #50 we ended with @pb-cdunn suggesting to try:

```
fc_run0 fc_run.cfg
```

Doing that now.

**Comment by JohnUrban** [comment]
Well, your `fc_run0` solution fixed that issue. So feel free to close this -- unless finding out if fc_run0 works is only the beginning of what you are looking to do.

```
#!/bin/bash
#SBATCH -J falcon-synth0-1
#SBATCH -c 4
#SBATCH --qos=ccmb-condo
#SBATCH --mem=4g
#SBATCH --time=00:30:00
###########################
unset PYTHONPATH
. ~/software/falcon/FALCON-integrate/fc_env/bin/activate
fc_run0 fc_run.cfg
```

If that is all, then I think I'm all set and ready to use 0.6.2.
Feel free to get in touch with me to do SLURM tests in the future. It is in my interest to make sure falcon stays compatible.

**Comment by pb-cdunn** [comment]
No, unfortunately `fc_run0` is deprecated. But you can use that to help us get `fc_run1` working. You should be able to find a log-file somewhere (several, actually) which shows the slurm command that was used to run a given task? Would you please tell me the full slurm command-line?

If you can run `fc_run1` in a different directory, then you will be able to compare. The slurm command is wrong for that.

**Comment by JohnUrban** [comment]
Do you know which log I should look in?
I grep'd everything in the dir and subdirs for "sbatch" -- I'd assume that is how falcon is submitting jobs to slurm. I did not find anything.

**Comment by pb-cdunn** [comment]
Maybe in `sge_logs/`? But that is for after the jobs are submitted. We want to see _how_ they were submitted. You should have a main `.log` file in your main run-dir, from Python logging. If not, then copy the `logging.ini` from **FALCON-examples** repo. You can provide that as an argument to `fc_run`. It's just standard Python logging-config format.

**Comment by JohnUrban** [comment]
I guess the only places I see sbatch in FALCON-integrate are:

```
./FALCON/falcon_kit/mains/run0.py:        sge_cmd="sbatch -J {job_name} {sge_option} {script}".format(job_name=job_name, cwd=os.getcwd(),sge_option=sge_option, script=script_fn)

./pypeFLOW/pwatcher/fs_based.py:        sge_cmd = 'sbatch -J {job_name} {sge_option} {specific} -D {cwd} -o stdout -e stderr -S {exe} {script_fn}'.format(
    **locals())

```

**Comment by pb-cdunn** [comment]
Exactly. And we'd like to see those in log-files, so we know exactly what was called.

**Comment by JohnUrban** [comment]
Ok.

I am using this logging.ini file:

```
[loggers]
keys=root,pypeflow,fc_run

[handlers]
keys=stream,file_pypeflow,file_fc_run,file_all

[formatters]
keys=form01,form02

[logger_root]
level=NOTSET
handlers=stream,file_all

[logger_pypeflow]
level=NOTSET
handlers=file_pypeflow
qualname=pypeflow
propagate=1

[logger_pwatcher]
level=NOTSET
handlers=file_pwatcher
qualname=pwatcher
propagate=1

[logger_fc_run]
level=NOTSET
handlers=file_fc_run
qualname=fc_run
propagate=1

[handler_stream]
class=StreamHandler
level=INFO
formatter=form02
args=(sys.stderr,)

[handler_file_pypeflow]
class=FileHandler
level=DEBUG
formatter=form01
args=('pypeflow.log',)

[handler_file_pwatcher]
class=FileHandler
level=DEBUG
formatter=form01
args=('pwatcher.log',)

[handler_file_fc_run]
class=FileHandler
level=DEBUG
formatter=form01
args=('fc_run.log',)

[handler_file_all]
class=FileHandler
level=DEBUG
formatter=form01
args=('fc.log',)

[formatter_form01]
format=%(asctime)s - %(name)s:%(lineno)d - %(levelname)s - %(message)s

[formatter_form02]
format=[%(levelname)s] %(message)s
```

With this batch script now:

```
#!/bin/bash
#SBATCH -J falcon-synth0-1
#SBATCH -c 4
#SBATCH --qos=ccmb-condo
#SBATCH --mem=4g
#SBATCH --time=00:30:00
###########################
unset PYTHONPATH
. ~/software/falcon/FALCON-integrate/fc_env/bin/activate
fc_run0 fc_run.cfg logging.ini
``
```

**Comment by pb-cdunn** [comment]
Good. You should see stuff both on the screen when you run, and in `fc.log`.

**Comment by JohnUrban** [comment]
What I see so far:

For fc_run0

```
$ grep -r sbatch . | awk 'match($0, /sbatch.*sh/) {print substr($0, RSTART, RLENGTH)}'

sbatch -J prepare_rdb.sh-task_build_rdb-task_build_rdb --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/prepare_rdb.sh
sbatch -J prepare_rdb.sh-task_build_rdb-task_build_rdb --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/prepare_rdb.sh
sbatch -J rj_0001.sh-d_0001_raw_reads-d_0001_raw_reads --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/job_0001/rj_0001.sh
sbatch -J rj_0000.sh-d_0000_raw_reads-d_0000_raw_reads --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/job_0000/rj_0000.sh
sbatch -J rj_0000.sh-d_0000_raw_reads-d_0000_raw_reads --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/job_0000/rj_0000.sh
sbatch -J rj_0001.sh-d_0001_raw_reads-d_0001_raw_reads --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/job_0001/rj_0001.sh
sbatch -J rp_00002.sh-m_00002_raw_reads-m_00002_raw_reads --cpus-per-task 2 --mem 10g --time 00:05:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/m_00002/rp_00002.sh
sbatch -J rp_00001.sh-m_00001_raw_reads-m_00001_raw_reads --cpus-per-task 2 --mem 10g --time 00:05:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/m_00001/rp_00001.sh
sbatch -J rp_00002.sh-m_00002_raw_reads-m_00002_raw_reads --cpus-per-task 2 --mem 10g --time 00:05:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/m_00002/rp_00002.sh
sbatch -J rp_00001.sh-m_00001_raw_reads-m_00001_raw_reads --cpus-per-task 2 --mem 10g --time 00:05:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/m_00001/rp_00001.sh
sbatch -J c_00002.sh-ct_00002-ct_00002 --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/preads/c_00002.sh
sbatch -J c_00001.sh-ct_00001-ct_00001 --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/preads/c_00001.sh
sbatch -J c_00002.sh-ct_00002-ct_00002 --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/preads/c_00002.sh
sbatch -J c_00001.sh-ct_00001-ct_00001 --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/preads/c_00001.sh
sbatch -J prepare_rdb.sh-task_build_rdb-task_build_rdb --cpus-per-task 8 --mem 10g --time 00:20:00 /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62/0-rawreads/prepare_rdb.sh
```

for fc_run1

```
sbatch -J J433327dc3c7d85 --cpus-per-task 8 --mem 10g --time 00:20:00 -V -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 -o stdout -e stderr -S /bin/bash /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/wrappers/run-J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875.bash
sbatch -J J433327dc3c7d85 --cpus-per-task 8 --mem 10g --time 00:20:00 -V -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 -o stdout -e stderr -S /bin/bash /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/wrappers/run-J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875.bash
```

I would take out "-o stdout -e stderr" as without those, both will be collapsed into a slurm-JOBID.out file.
Alternatively, you could just use "-o specific-name-for-job.out" and both stdout and stderr will go to that file. 
I am not sure what "-S /bin/bash" is doing….

```
-S, --core-spec=cores       count of reserved cores
```

**Comment by JohnUrban** [comment]
This is probably the problem:

```
-V, --version               output version information and exit
```

**Comment by pb-cdunn** [comment]
Thanks for catching it.

Could you provide a link to slurm docs which correspond to what you are using? Maybe I looked at the wrong web-page.

**Comment by JohnUrban** [comment]
This can probably be:

```
./pypeFLOW/pwatcher/fs_based.py:        
sge_cmd = 'sbatch -J {job_name} {sge_option} -D {cwd} {script_fn}'.format(
    **locals())
```

**Comment by JohnUrban** [comment]
Apologies for the dump here….

sbatch -h

```
Usage: sbatch [OPTIONS...] executable [args...]

Parallel run options:
  -a, --array=indexes         job array index values
  -A, --account=name          charge job to specified account
      --bb=<spec>             burst buffer specifications
      --begin=time            defer job until HH:MM MM/DD/YY
  -M, --clusters=names        Comma separated list of clusters to issue
                              commands to.  Default is current cluster.
                              Name of 'all' will submit to run on all clusters.
      --comment=name          arbitrary comment
      --cpu-freq=min[-max[:gov]] requested cpu frequency (and governor)
  -c, --cpus-per-task=ncpus   number of cpus required per task
  -d, --dependency=type:jobid defer job until condition on jobid is satisfied
  -D, --workdir=directory     set working directory for batch script
  -e, --error=err             file for batch script's standard error
      --export[=names]        specify environment variables to export
      --export-file=file|fd   specify environment variables file or file
                              descriptor to export
      --get-user-env          load environment from local cluster
      --gid=group_id          group ID to run job as (user root only)
      --gres=list             required generic resources
  -H, --hold                  submit job in held state
      --ignore-pbs            Ignore #PBS options in the batch script
  -i, --input=in              file for batch script's standard input
  -I, --immediate             exit if resources are not immediately available
      --jobid=id              run under already allocated job
  -J, --job-name=jobname      name of job
  -k, --no-kill               do not kill job on node failure
  -L, --licenses=names        required license, comma separated
  -m, --distribution=type     distribution method for processes to nodes
                              (type = block|cyclic|arbitrary)
      --mail-type=type        notify on state change: BEGIN, END, FAIL or ALL
      --mail-user=user        who to send email notification for job state
                              changes
  -n, --ntasks=ntasks         number of tasks to run
      --nice[=value]          decrease scheduling priority by value
      --no-requeue            if set, do not permit the job to be requeued
      --ntasks-per-node=n     number of tasks to invoke on each node
  -N, --nodes=N               number of nodes on which to run (N = min[-max])
  -o, --output=out            file for batch script's standard output
  -O, --overcommit            overcommit resources
  -p, --partition=partition   partition requested
      --parsable              outputs only the jobid and cluster name (if present),
                              separated by semicolon, only on successful submission.
      --power=flags           power management options
      --priority=value        set the priority of the job to value
      --profile=value         enable acct_gather_profile for detailed data
                              value is all or none or any combination of
                              energy, lustre, network or task
      --propagate[=rlimits]   propagate all [or specific list of] rlimits
      --qos=qos               quality of service
  -Q, --quiet                 quiet mode (suppress informational messages)
      --reboot                reboot compute nodes before starting job
      --requeue               if set, permit the job to be requeued
  -s, --share                 share nodes with other jobs
  -S, --core-spec=cores       count of reserved cores
      --sicp                  If specified, signifies job is to receive
      --signal=[B:]num[@time] send signal when time limit within time seconds
      --switches=max-switches{@max-time-to-wait}
                              Optimum switches and max time to wait for optimum
      --thread-spec=threads   count of reserved threads
  -t, --time=minutes          time limit
      --time-min=minutes      minimum time limit (if distinct)
      --uid=user_id           user ID to run job as (user root only)
  -v, --verbose               verbose mode (multiple -v's increase verbosity)
      --wckey=wckey           wckey to run job under
      --wrap[=command string] wrap command string in a sh script and submit

Constraint options:
      --contiguous            demand a contiguous range of nodes
  -C, --constraint=list       specify a list of constraints
  -F, --nodefile=filename     request a specific list of hosts
      --mem=MB                minimum amount of real memory
      --mincpus=n             minimum number of logical processors (threads)
                              per node
      --reservation=name      allocate resources from named reservation
      --tmp=MB                minimum amount of temporary disk
  -w, --nodelist=hosts...     request a specific list of hosts
  -x, --exclude=hosts...      exclude a specific list of hosts

Consumable resources related options:
      --exclusive[=user]      allocate nodes in exclusive mode when
                              cpu consumable resource is enabled
      --mem-per-cpu=MB        maximum amount of real memory per allocated
                              cpu required by the job.
                              --mem >= --mem-per-cpu if --mem is specified.

Affinity/Multi-core options: (when the task/affinity plugin is enabled)
  -B  --extra-node-info=S[:C[:T]]            Expands to:
       --sockets-per-node=S   number of sockets per node to allocate
       --cores-per-socket=C   number of cores per socket to allocate
       --threads-per-core=T   number of threads per core to allocate
                              each field can be 'min' or wildcard '*'
                              total cpus requested = (N x S x C x T)

      --ntasks-per-core=n     number of tasks to invoke on each core
      --ntasks-per-socket=n   number of tasks to invoke on each socket


Help options:
  -h, --help                  show this help message
  -u, --usage                 display brief usage message

Other options:
  -V, --version               output version information and exit


```

As for a link -- this is from `man slurm`:
http://slurm.schedmd.com/

For sbatch specifically:
http://slurm.schedmd.com/sbatch.html

Seems consistent with a very very quick glance … i.e. I only looked at -V so far.

**Comment by pb-cdunn** [comment]
`-V` was a mistake. That's for **Torque**, not Slurm. (We can't easily test Torque either, so we just copy/paste from users.)

**Comment by pb-cdunn** [comment]
We also need `--export`. No problem.

But I don't see a way to specify the shell (`bash`). Does Slurm depend on shebang lines?

**Comment by pb-cdunn** [comment]
Oh, I see. We could use `--wrap` if we don't want shebangs.

```
--wrap[=command string] wrap command string in a sh script and submit
```

Well, shebang is fine for now. (The issue is that the script might not yet be executable in the filesystem, because of spurious filesystem delays. But we'll deal with that if it comes up.)

**Comment by pb-cdunn** [comment]
Could you `git pull origin` in pypeFLOW and try again (using `fc_run1`)? You'll want this commit:

```
commit d8e24efc9927d9222b3d33ca4e9c9c5f89fec152
Author: Christopher Dunn <cdunn@pacificbiosciences.com>
Date:   Thu Jun 2 15:32:53 2016

    Fixes for Slurm
```

**Comment by JohnUrban** [comment]
I don't think the git command worked for me:

```
$ git pull origin
remote: Counting objects: 4, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 4 (delta 0), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (4/4), done.
From git://github.com/PacificBiosciences/pypeFLOW
   d2a3a4d..d8e24ef  master     -> origin/master
You asked to pull from the remote 'origin', but did not specify
a branch. Because this is not the default configured remote
for your current branch, you must specify a branch on the command line.
```

I didnt see any change to:

```
ge_cmd = 'sbatch -J {job_name} {sge_option} {specific} -D {cwd} -o stdout -e stderr -S {exe} {script_fn}'.format(
Binary file ./pwatcher/fs_based.pyc matches
```

And the same problem persisted...

**Comment by JohnUrban** [comment]
I just made the change I suggested above and it is working fine.

```
./pypeFLOW/pwatcher/fs_based.py:        
sge_cmd = 'sbatch -J {job_name} {sge_option} -D {cwd} {script_fn}'.format(
    **locals())
```

It finished the synth0 assembly successfully.

…no other change needed apparently…?

**Comment by pb-cdunn** [comment]
If your repo is on the `master` branch, you would need `git pull origin master`.

> …no other change needed apparently…?

My changes will end up in the official release. I cannot test Slurm, so if you don't test it for me, then it might break you later. There's not much I can do about that.

The problem with your change is that it depends on the file actually existing as an executable. Just because it works for you does not mean it will work in other filesystems, where there may be significant latencies. These filesystem issues drive me crazy, so I have to be very conservative about them. That's why it's important to run the script under a shell, rather than as an executable. That's why I want to use `--wrap`.

**Comment by JohnUrban** [comment]
No problem. I will test your code.

**Comment by JohnUrban** [comment]
The update did not work. It crashed with error messages. Let me know what you want to see and I will grab it for you.

**Comment by pb-cdunn** [comment]
I'm not sure where to look. We need the stderr from the sbatch command, and the sbatch command itself. I think the command itself will be in `all.log`. Actually, just try to repeat the sbatch command from your shell; that will probably repeat the message.

**Comment by JohnUrban** [comment]
```
$ sbatch -J J433327dc3c7d85 --cpus-per-task 8 --mem 10g --time 00:20:00 --export -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 -o stdout -e stderr --wrap="/bin/bash /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/wrappers/run-J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875.bash"
sbatch: error: Batch script is empty!
```

If I do:

```
$ /bin/bash /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/wrappers/run-J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875.bash
```

It generates a bunch of output….

Was able to get it to work as:

```
sbatch -J J433327dc3c7d85 --cpus-per-task 8 --mem 10g --time 00:20:00 -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 --wrap="/path/to/script.bash"
```

and

```
sbatch -J J433327dc3c7d85-2 --cpus-per-task 8 --mem 10g --time 00:20:00 -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 --wrap="/bin/bash /path/to/script.bash"
```

and

```
sbatch -J J433327dc3c7d85-2 --cpus-per-task 8 --mem 10g --time 00:20:00 -D /gpfs/scratch/jurban/male-ilmn/lambda/falcon/synth0/FALCON62_1/mypwatcher/jobs/J433327dc3c7d8583cd6568eaae6915aa4bb7c5254cdb76dd6500af649890e875 -o stdout -e stderr --wrap="/bin/bash /path/to/script.bash"
```

So the only offending flag was --export

**Comment by pb-cdunn** [comment]
What do you suggest we do about the environment? Assume it will be fine? I don't think that will fly. When you distribute our tools, we wrap each one in its own environment. We definitely need to pass _something_ along to the batch job.

If `--export` does not work, and if you cannot find an alternative, then we will have to dump our ENV into our bash scripts.

> --export[=names]        specify environment variables to export

**Comment by JohnUrban** [comment]
It is a good question. It looks like the default behavior is to propagate all variables in the environment (without specifying --export).

Maybe this will be helpful - particularly, "By default all environment variables are propagated."
http://slurm.schedmd.com/sbatch.html

```
--export=<environment variables | ALL | NONE>
Identify which environment variables are propagated to the batch job. Multiple environment variable names should be comma separated. Environment variable names may be specified to propagate the current value of those variables (e.g. "--export=EDITOR") or specific values for the variables may be exported (e.g.. "--export=EDITOR=/bin/vi") in addition to the environment variables that would otherwise be set. This option particularly important for jobs that are submitted on one cluster and execute on a different cluster (e.g. with different paths). By default all environment variables are propagated. If the argument is NONE or specific environment variable names, then the --get-user-env option will implicitly be set to load other environment variables based upon the user's configuration on the cluster which executes the job.
```

**Comment by JohnUrban** [comment]
I think specifying `--export` with nothing specified to export is resulting in exporting an empty environment… maybe. Either way, not including `--export` means all variables are exported.

**Comment by pb-cdunn** [comment]
Ok. Thx. Fixed on `master` branch of **pypeFLOW**. I will push out a new release after I merge some changes from thegenemyers.

**Comment by JohnUrban** [comment]
It looks like it is working.

**Comment by JohnUrban** [comment]
Well, I should be more clear. Updating pypeFLOW with what you have now (`git pull origin master`) definitely results in 0.6.2 working (using `fc_run1`). Synth0 assembled.

**Comment by pb-cdunn** [comment]
You can pull `master` from **FALCON-integrate** now if you want. That will be `0.6.3` soon, I think. You'd also get the latest DALIGNER.

**Comment by JohnUrban** [comment]
I saw a note in `pwatcher/fs_based.py` about capturing sbatch jobID. If you know the following already, please ignore.
 If you were to want the SLURM JobID on the stdout or stderr name, it would be: `-o %A.out -e %A.err` or more descriptive might be `-o {job_name}.%A.out`. Inside a script it is accessible by `$SLURM_JOBID`.

**Comment by pb-cdunn** [comment]
That's helpful. Thx.

I am fixing the `0.6.3` release...

**Comment by JohnUrban** [comment]
Yes - I saw that the 0.6.3 release page says there are fixes for Torque/LSF and was wondering if 0.6.3 would also be including the fixes for Slurm.
github.com/PacificBiosciences/FALCON-integrate/releases/tag/0.6.3

**Comment by pb-cdunn** [comment]
Yes, I think so. Maybe I should have said Slurm/LSF.
